ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeViewshed/AnalyzeViewshed.cpp

```diff
@@ -213,7 +213,8 @@ void AnalyzeViewshed::calculateViewshed()
 void AnalyzeViewshed::processResults(GeoprocessingResult *results)
 {
   // Get the results from the outputs as GeoprocessingFeatures
-  GeoprocessingFeatures* viewshedResultFeatures = static_cast<GeoprocessingFeatures*>(results->outputs()["Viewshed_Result"]);
```

Comment:
This is another example of using a non-const Qt's container with a non-const function. In this code, `results->outputs()` is not const then `operator[]` is not const and the container is detached (deep copy).

To solve that, it's necessary to force `const`:
- using `qAsConst`
- or creating a local const variable

The difficulty with `qAsConst` in this code is it must be used with the objet, not the container:
```cpp
... static_cast<GeoprocessingFeatures*>(qAsConst(results)->outputs()["Viewshed_Result"]);
// not
// ... static_cast<GeoprocessingFeatures*>(qAsConst(results->outputs())["Viewshed_Result"]);
```

I don't know if it's clear that `const` and `qAsConst` are used here to avoid to detach the container. Which syntaxe do you prefer?

Reply:
I like your proposed solution of creating a const local.